lib/compilers/rustc-cg-gcc.js

```diff
@@ -43,10 +50,27 @@ export class RustcCgGCCCompiler extends RustCompiler {
             '-Z', 'codegen-backend=librustc_codegen_gcc.so',
             '--sysroot', toolroot + '/sysroot' ];
 
-        // rust.js makes the asumption that LLVM is used. This may go away when cranelift is available.
-        // Until this is the case and the super() class is refactored, simply ditch -Cllvm arg.
+        // rust.js makes the asumption that LLVM is used. This may go away when
+        // cranelift is available. Until this is the case and the super() class
+        // is refactored, simply ditch -Cllvm-args options as the super class
+        // uses it for LLVM specific arg.
+        // Beware that the GCC backend still uses -C llvm-args as a way to
+        // get option from the user, so don't filter it too agressively.
         let super_options = super.optionsForFilter(filters, outputFilename, userOptions).filter(arg => !/-Cllvm.*/.test(arg));
         options = options.concat(super_options);
         return options;
     }
+    getDefaultExecOptions() {
+        const execOptions = super.getDefaultExecOptions();
+
+        // Override the TMPDIR as we may want to capture dump files.
+        // The dumps will be created by libgccjit in a subdirectory in $TMPDIR.
+        // Having a clean and private $TMPDIR makes the searching of dump files
+        // easier.
+        const tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'rustc-cg-gcc-'));
```

Comment:
I don't know if this will work in our sandbox environment where we only map some special directories in. Arbitrary new directories in `/tmp` (or `os.tmpdir()`) won't be mapped into our sandbox.

I wonder if we can use a subdirectory of the output directory - which is [a] guaranteed to be mapped and [b] we remap arguments to point at the sandbox. That said: we don't remap environment variables, so we'd need to update our exec.js to do that.

I also can't see how to get the output directory at this point. The `runCompiler` method knows this (it sets the `customCwd` which is essentially the output directory (and has all the magic exec.js remappings done).

I realise this is super tricky as it's nearly impossible to test locally. I can run it locally... so I think I can help here.

Reply:
I can only trust you here :). Would have been better if you didn't have to do more work. I can also see if there's something possible with `libgccjit`. That was my initial idea. GCC has options for tuning how dump files are created, but this is simply not playing well with the jit. Maybe I could try a little harder and ask for help from the jit devs. TMPDIR was really the easy solution, or so I thought :)

Reply:
Got it! Nothing's easy on Compiler Explorer I'm afraid :( I try but in particular this stuff is hard.

If the TMPDIR thing is still something we want to explore I can either help you get set up with nsjail (ping me on Discord), or else if you tell me what I should expect this PR to do (I need an idiot's guide - which compiler specifically, and then presumably I just enable the "ast" tool?!) I can try testing it locally with nsjail etc and fix it there.